backend/LexBoxApi/Controllers/LoginController.cs

```diff
@@ -36,6 +41,35 @@ public class LoginController : ControllerBase
         return Redirect(returnTo);
     }
 
+    [HttpGet("verifyEmail")]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    public async Task<ActionResult<LexAuthUser>> VerifyEmail(
+        string jwt, // This is required because auth looks for a jwt in the query string
+        string returnTo,
+        string token,
+        [EmailAddress] string email)
+    {
+        var userId = _loggedInContext.User.Id;
+        var user = _lexBoxDbContext.Users.Find(userId);
+        if (user == null) return Unauthorized();
+
+        if (token == user.EmailVerificationToken && email == user.Email)
```

Comment:
instead of the token check, use `loggedInContext.User.EmailVerificationRequired != true`